src/board/shell.js

```diff
@@ -379,12 +378,12 @@ export default class Shell {
     var command =
         "import uhashlib,ubinascii,sys\r\n" +
         "hash = uhashlib.sha256()\r\n" +
-        "f = open('"+filename+"', 'rb')\r\n"+
-        "while True:\r\n"+
-        '    c = f.read('+this.BIN_CHUNK_SIZE+')\r\n'+
-        '    if not c:\r\n'+
-        '        break\r\n'+
-        '    hash.update(c)\r\n'+
+        "with open('"+filename+"', 'rb') as f:\r\n"+
```

Comment:
That's indeed a lot nicer! But does this effect memory on the long run? I get that the scope of 'f' is now smaller and the file gets closed properly, but if the variable 'f' is overwritten from the last command like it was doing before this change, it's also removing (and garbage collecting) the previous contents of 'f', right? Just curious why this works.